packages/repository-tests/src/crud/relations/integration/has-many.factory.integration.ts

```diff
@@ -0,0 +1,259 @@
+      expect(toJSON(orders)).to.not.containEql(toJSON(notMyOrder));
+    });
+
+    it('finds appropriate related model instances for multiple relations', async () => {
+      // note(shimks): roundabout way of creating reviews with 'approves'
+      // ideally, the review repository should have a approve function
+      // which should 'approve' a review
+      // On another note, this test should be separated for 'create' and 'find'
+      await customerAuthoredReviewFactoryFn(existingCustomerId).create({
+        description: 'my wonderful review',
+        approvedId: existingCustomerId + 1,
+      });
+      await customerAuthoredReviewFactoryFn(existingCustomerId + 1).create({
+        description: 'smash that progenitor loving approve button',
+        approvedId: existingCustomerId,
+      });
+
+      const reviewsApprovedByCustomerOne = await customerApprovedReviewFactoryFn(
+        existingCustomerId,
+      ).find();
+      const reviewsApprovedByCustomerTwo = await customerApprovedReviewFactoryFn(
+        existingCustomerId + 1,
+      ).find();
+
+      const persistedReviewsApprovedByCustomerOne = await reviewRepo.find({
+        where: {
+          approvedId: existingCustomerId,
+        },
+      });
+      const persistedReviewsApprovedByCustomerTwo = await reviewRepo.find({
+        where: {
+          approvedId: existingCustomerId + 1,
+        },
+      });
+
+      expect(reviewsApprovedByCustomerOne).to.eql(
+        persistedReviewsApprovedByCustomerOne,
+      );
+      expect(reviewsApprovedByCustomerTwo).to.eql(
+        persistedReviewsApprovedByCustomerTwo,
+      );
+    });
+
+    //--- HELPERS ---//
+
+    async function givenPersistedCustomerInstance() {
+      const customer = await customerRepo.create({name: 'a customer'});
+      existingCustomerId = customer.id;
+    }
+
+    function givenConstrainedRepositories() {
+      const orderFactoryFn = createHasManyRepositoryFactory<
+        Order,
+        typeof Order.prototype.id,
+        typeof Customer.prototype.id
+      >(
+        Customer.definition.relations.orders as HasManyDefinition,
+        Getter.fromValue(orderRepo),
+      );
+
+      customerOrderRepo = orderFactoryFn(existingCustomerId);
+    }
+
+    function givenRepositoryFactoryFunctions() {
+      customerAuthoredReviewFactoryFn = createHasManyRepositoryFactory(
+        Customer.definition.relations.reviewsAuthored as HasManyDefinition,
+        Getter.fromValue(reviewRepo),
+      );
+      customerApprovedReviewFactoryFn = createHasManyRepositoryFactory(
+        Customer.definition.relations.reviewsApproved as HasManyDefinition,
+        Getter.fromValue(reviewRepo),
+      );
+    }
+  });
+
+  function givenCrudRepositories() {
+    customerRepo = new repositoryClass(Customer, db);
+    orderRepo = new repositoryClass(Order, db);
+    reviewRepo = new repositoryClass(Review, db);
+  }
+
+  //--- HELPERS ---//
+
+  class Order extends Entity {
+    id: string;
```

Comment:
I think this should be using `mixedIdType`?

On the second thought, do we really care about the actual type, considering that it's a union `number | string` and therefore we can do very little with those values? What if we used `unknown` everywhere we need to describe the type of a PK or a FK property?